ding/config/config.py

```diff
@@ -3,11 +3,11 @@
 import json
 import shutil
 import sys
-import time
 import tempfile
-from importlib import import_module
-from typing import Optional, Tuple, NoReturn
+import warnings
```

Comment:
Don't use `warnings` lib, it would lead to irregular terminal output, `logging.warning` can be better choice.